File: src/se/axelbengtsson/kinapokerscorecalculator/Round.java
package se.axelbengtsson.kinapokerscorecalculator;

public class Round {
  final int numberOfPlayers;
  final Player[] players;
  final PlayerRound[] playerRound;

  public Round(int numberOfPlayers) {
    this.numberOfPlayers = numberOfPlayers;
    this.players = new Player[numberOfPlayers];
    this.playerRound = new PlayerRound[numberOfPlayers];
    if (numberOfPlayers < 0 || numberOfPlayers > 4) {
      throw new RuntimeException("Number of player can only be 0 - 4");
    }
    for( Player p : Player.values()) {
      if (p.isPlaying(numberOfPlayers)) {
        this.players[p.index(numberOfPlayers)] = p;
      }
    }
  }
}

File: app/src/main/java/se/axel/bengtsson/kinapokerscorecalculator/Round.kt
package se.axel.bengtsson.kinapokerscorecalculator

import java.lang.RuntimeException

class Round(val numberOfPlayers: Int) {

  val players: Array<Player>
  val playerRound: Array<PlayerRound>

  init {
    // Validation
    if (numberOfPlayers < 0 || numberOfPlayers > 4) {
      throw RuntimeException("Number of player can only be 0 - 4")
    }
    // Initiate
    players = Player.values().filter { player -> player.isPlaying(numberOfPlayers) }.toTypedArray()
    playerRound = players.map { PlayerRound(it) }.toTypedArray()
  }
}
